smartcontract/service/native/cross_chain/lock_proxy/lock_proxy.go

```diff
@@ -0,0 +1,376 @@
+		return utils.BYTE_FALSE, fmt.Errorf("[BindAssetHash] checkWitness error: %s", err)
+	}
+	// store the target asset hash
+	native.CacheDB.Put(GenBindAssetHashKey(contract, bindParam.SourceAssetHash, bindParam.TargetChainId), utils.GenVarBytesStorageItem(bindParam.TargetAssetHash).ToArray())
+
+	// make sure the new limit is greater than the stored limit
+	limitKey := GenCrossedLimitKey(contract, bindParam.SourceAssetHash, bindParam.TargetChainId)
+	limit, err := getAmount(native, limitKey)
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("[BindAssetHash] getCrossedLimit error:%s", err)
+	}
+	if bindParam.Limit.Cmp(limit) != 1 {
+		return utils.BYTE_FALSE, fmt.Errorf("[BindAssetHash] new Limit:%s should be greater than stored Limit:%s", bindParam.Limit.String(), limit.String())
+	}
+	// if the source asset hash is the target chain asset, increase the crossedAmount value by the limit increment
+	if bindParam.IsTargetChainAsset {
+		increment := big.NewInt(0).Sub(bindParam.Limit, limit)
+		crossedAmountKey := GenCrossedAmountKey(contract, bindParam.SourceAssetHash, bindParam.TargetChainId)
+		crossedAmount, err := getAmount(native, crossedAmountKey)
+		if err != nil {
+			return utils.BYTE_FALSE, fmt.Errorf("[BindAssetHash] getCrossedAmount error:%s", err)
+		}
+		newCrossedAmount := big.NewInt(0).Add(crossedAmount, increment)
+		if newCrossedAmount.Cmp(crossedAmount) != 1 {
+			return utils.BYTE_FALSE, fmt.Errorf("[BindAssetHash] new crossedAmount:%s is not greater than stored crossed amount:%s", newCrossedAmount.String(), crossedAmount.String())
+		}
+		native.CacheDB.Put(crossedAmountKey, utils.GenVarBytesStorageItem(newCrossedAmount.Bytes()).ToArray())
+	}
+	// update the new limit
+	native.CacheDB.Put(GenCrossedLimitKey(contract, bindParam.SourceAssetHash, bindParam.TargetChainId), utils.GenVarBytesStorageItem(bindParam.Limit.Bytes()).ToArray())
+	if config.DefConfig.Common.EnableEventLog {
+		native.Notifications = append(native.Notifications,
+			&event.NotifyEventInfo{
+				ContractAddress: contract,
+				States:          []interface{}{BIND_ASSET_NAME, hex.EncodeToString(bindParam.SourceAssetHash[:]), bindParam.TargetChainId, hex.EncodeToString(bindParam.TargetAssetHash), bindParam.Limit.String()},
+			})
+	}
+	return utils.BYTE_TRUE, nil
+}
+
+func Lock(native *native.NativeService) ([]byte, error) {
+	contract := native.ContextRef.CurrentContext().ContractAddress
+	ontContract := utils.OntContractAddress
+	ongContract := utils.OngContractAddress
+	source := common.NewZeroCopySource(native.Input)
+
+	var lockParam LockParam
+	if err := lockParam.Deserialization(source); err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("[Lock] contract params deserialization error:%s", err)
+	}
+
+	if lockParam.Value == 0 {
+		return utils.BYTE_FALSE, nil
+	}
+	// currently, only support ont and ong lock operation
+	if !bytes.Equal(lockParam.SourceAssetHash[:], ontContract[:]) && !bytes.Equal(lockParam.SourceAssetHash[:], ongContract[:]) {
```

Comment:
you can use `lockParam.SourceAssetHash != ontContract`

Reply:
updated